var/spack/repos/builtin/packages/zoltan/package.py

```diff
@@ -143,19 +155,26 @@ def configure_args(self):
                 config_args.extend(["FC={0}".format(spec["mpi"].mpifc)])
 
         config_fcflags = config_cflags[:]
+        config_cxxflags = config_cflags[:]
+
         if spec.satisfies("%gcc@10:+fortran"):
             config_fcflags.append("-fallow-argument-mismatch")
+
         # NOTE: Early versions of Zoltan come packaged with a few embedded
         # library packages (e.g. ParMETIS, Scotch), which messes with Spack's
         # ability to descend directly into the package's source directory.
-        config_args.extend(
-            [
-                "--with-cflags={0}".format(" ".join(config_cflags)),
-                "--with-cxxflags={0}".format(" ".join(config_cflags)),
-                "--with-fcflags={0}".format(" ".join(config_fcflags)),
-                "--with-ldflags={0}".format(" ".join(config_ldflags)),
-            ]
-        )
+        if config_cflags:
+            config_args.append("--with-cflags={0}".format(" ".join(config_cflags)))
+        if config_cxxflags:
+            config_args.append("--with-cxxflags={0}".format(" ".join(config_cflags)))
```

Comment:
I think there is a typo here:
```suggestion
        if config_cxxflags:
            config_args.append("--with-cxxflags={0}".format(" ".join(config_cxxflags)))
```
Also, this can possibly be removed to save a copy of `cflags` and a couple of lines here:
```python
if config_cflags:
    config_args.append("--with-cflags={0}".format(" ".join(config_cflags)))
    config_args.append("--with-cxxflags={0}".format(" ".join(config_cflags)))
```